homeassistant/components/freedompro/switch.py

```diff
@@ -0,0 +1,93 @@
+"""Support for Freedompro switch."""
+import json
+
+from pyfreedompro import put_state
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Freedompro switch."""
+    api_key = entry.data[CONF_API_KEY]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        Device(hass, api_key, device, coordinator)
+        for device in coordinator.data
+        if device["type"] == "switch" or device["type"] == "outlet"
+    )
+
+
+class Device(CoordinatorEntity, SwitchEntity):
+    """Representation of an Freedompro switch."""
+
+    def __init__(self, hass, api_key, device, coordinator):
+        """Initialize the Freedompro switch."""
+        super().__init__(coordinator)
+        self._hass = hass
+        self._session = aiohttp_client.async_get_clientsession(self._hass)
+        self._api_key = api_key
+        self._attr_name = device["name"]
+        self._attr_unique_id = device["uid"]
+        self._type = device["type"]
```

Comment:
`self._type` isn't used outside of init method. It can be a local variable instead.